osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/infra/jms/protocol/outbound/ProtocolRequestMessageSender.java

```diff
@@ -68,50 +65,45 @@ private void sendMessage(final ProtocolRequestMessage requestMessage, final Prot
             final JmsTemplate jmsTemplate) {
         LOGGER.info("Sending request message to protocol requests queue");
 
-        final Long originalTimeToLive = jmsTemplate.getTimeToLive();
+        final long originalTimeToLive = jmsTemplate.getTimeToLive();
         boolean isCustomTimeToLiveSet = false;
         if (requestMessage.getMessageType().equals(GET_POWER_USAGE_HISTORY.toString())) {
             jmsTemplate.setTimeToLive(this.getPowerUsageHistoryRequestTimeToLive);
             isCustomTimeToLiveSet = true;
         }
 
-        jmsTemplate.send(new MessageCreator() {
-
-            @Override
-            public Message createMessage(final Session session) throws JMSException {
-
-                final String deviceIdentification = requestMessage.getDeviceIdentification();
-
-                final ObjectMessage objectMessage = session.createObjectMessage(requestMessage.getRequest());
-                objectMessage.setJMSCorrelationID(requestMessage.getCorrelationUid());
-                objectMessage.setJMSType(requestMessage.getMessageType());
-                objectMessage.setJMSPriority(requestMessage.getMessagePriority());
-                objectMessage.setStringProperty(Constants.DOMAIN, requestMessage.getDomain());
-                objectMessage.setStringProperty(Constants.DOMAIN_VERSION, requestMessage.getDomainVersion());
-                objectMessage.setStringProperty(Constants.ORGANISATION_IDENTIFICATION,
-                        requestMessage.getOrganisationIdentification());
-                objectMessage.setStringProperty(Constants.DEVICE_IDENTIFICATION, deviceIdentification);
-                objectMessage.setStringProperty(Constants.IP_ADDRESS, requestMessage.getIpAddress());
-                objectMessage.setBooleanProperty(Constants.IS_SCHEDULED, requestMessage.isScheduled());
-                objectMessage.setIntProperty(Constants.RETRY_COUNT, requestMessage.getRetryCount());
-                objectMessage.setBooleanProperty(Constants.BYPASS_RETRY, requestMessage.bypassRetry());
-
-                if (!protocolInfo.isParallelRequestsAllowed()) {
-                    final String messageGroupId = ProtocolRequestMessageSender.this
-                            .getMessageGroupId(deviceIdentification);
-                    LOGGER.debug("Setting message group property for device {} to: {}", deviceIdentification,
-                            messageGroupId);
-                    objectMessage.setStringProperty(Constants.MESSAGE_GROUP, messageGroupId);
-                }
-
-                return objectMessage;
+        jmsTemplate.send(session -> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this lambda expression number of lines from 23 to at most 20. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1188)